### ISSUE TITLE
2016.12.15.1 still hangs/crashes Firefox 50.1.0

### ISSUE THREAD

**Issue by Newrad0603** [EFForg/privacybadger#1117]
I uninstalled Privacy Badger previously because it caused Firefox to hang/crash. Today, I saw that a newer version had been pushed, so I installed it, and it caused my Firefox process to peg a CPU core, hung the process, and started growing in memory until the process crashed. Subsequent launches would not show any UI, but a core would be pegged and the memory would grow until Firefox crashed. Disabling PB allows Firefox to start up, but then enabling it during a running session immediately causes a hang and the same issues that lead to Firefox crashing.

**Comment by cooperq** [comment]
If you create a fresh firefox profile and install privacy badger do you still have this issue?

**Comment by kehugter** [comment]
Privacy Badger 2016.12.15.1
Firefox 50.1.0 from Debian repos
Debian testing
Linux 4.8.0-2-amd64

Having the same issue. Firefox starts, puts a 10-15% load on the CPU and fills the memory at ~40mbps. Disabled all other extensions, issue persists.

I can start Firefox __once__ with the following procedure:
- disable Privacy Badger from `--safe-mode`
- restart Firefox
- enable Privacy Badger, Firefox hangs again
- kill and start Firefox again

> fresh firefox profile [...] still have this issue?

No, it works. I can enable and disable Privacy Badger too without hangs.

---

Startup log

```
1483613486694	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ@jetpack': Reading manifest: Error processing permissions.9: Unknown permission "privacy"
1483613486697	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ@jetpack': Reading manifest: Error processing author: An unexpected property was found in the WebExtension manifest.
1483613486697	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ@jetpack': Reading manifest: Error processing minimum_chrome_version: An unexpected property was found in the WebExtension manifest.
1483613486697	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ@jetpack': Reading manifest: Error processing options_page: An unexpected property was found in the WebExtension manifest.
1483613486727	addons.webextension.jid1-MnnxcxisBPnSXQ@jetpack	WARN	Please specify whether you want browser_style or not in your browser_action options.
```

the only warnings are related to Privacy Badger, but those appear in both the old and the new profile.

**Comment by cooperq** [comment]
yea those warnings are fine. I think the issue is with privacy badger trying to migrate your old data. I bet you have a lot of domains in your profile! 

Does deleteing privacy badger entirely, restarting firefox, and then reinstalling privacy badger fix the issue?

**Comment by kehugter** [comment]
> I bet you have a lot of domains in your profile!

This ones? Yes, they look quite long and duplicated. Is that a result of the failed migration?

```
$ ls -l browser-extension-data/jid1-MnnxcxisBPnSXQ@jetpack/storage.js jetpack/jid1-MnnxcxisBPnSXQ@jetpack/simple-storage/store.json
-rw------- 1 pmk pmk 224780 Jan  2 12:37 browser-extension-data/jid1-MnnxcxisBPnSXQ@jetpack/storage.js
-rw-r--r-- 1 pmk pmk 224780 Jan  2 12:37 jetpack/jid1-MnnxcxisBPnSXQ@jetpack/simple-storage/store.json
```

```
$ python -m json.tool browser-extension-data/jid1-MnnxcxisBPnSXQ@jetpack/storage.js | wc -l
12294
```

> Does deleteing privacy badger entirely, restarting firefox, and then reinstalling privacy badger fix the issue?

Nope, hangs after installing.

**Comment by Newrad0603** [comment]
Creating a new profile worked for me as well, but I'd prefer to to have to use that solution :)

I have tried uninstalling PrivacyBadger and reinstalling it many times on the existing profile to no avail (still hangs). I did setup up a fair number of custom settings for various sites. Is there cached data somewhere that can be deleted that is being left behind when Firefox uninstalls the extension?

**Comment by cooperq** [comment]
deleting both of the files that @kehugter mentions should do the trick. Note that this will delete all of your privacy badger domains and settings, so it will effectively completetly reset privacy badger but if it works again maybe that is an acceptable trade-off.

**Comment by Newrad0603** [comment]
That seems to have done the trick. I can now use FF and PB again! Thanks for the help.

**Comment by kehugter** [comment]
Closed? PB may still crash browsers in the future, expecting users to delete a random file in the profile shouldn't be the fix.

**Comment by kehugter** [comment]
^ @cooperq

**Comment by cooperq** [comment]
@kehugter good point, thank you. I think we have figured out the source of the slowness and should have a fix in the next build. See #1145

**Comment by ghostwords** [comment]
Browser lockups should be fixed by (some combination of) #1082, #1145 and #1182.

**Comment by kehugter** [comment]
Tested with my old `storage.js`/`store.json` file, can confirm that the issue is fixed. Thanks!
